library/kani_macros/src/sysroot/contracts.rs

```diff
@@ -120,7 +120,18 @@ where
         && path.segments.iter().zip(mtch).all(|(actual, expected)| actual.ident == *expected)
 }
 
-/// Classifies the state a function is in the contract handling pipeline.
+/// Temporarily swap `$src` and `$target` using `std::mem::swap` for the
+/// execution of `$code`, then swap them back.
+macro_rules! swapped {
+    ($src:expr, $target:expr, $code:expr) => {{
+        std::mem::swap($src, $target);
+        let result = $code;
```

Comment:
Can you please add the type here? I got the following error:
```rust
error[E0282]: type annotations needed
  --> external_contract.rs:28:17
   |
28 | #[kani::ensures(result.is_some() && rectangle.can_hold(result.unwrap()))]
   |                 ^^^^^^ cannot infer type

```

Reply:
Can you provide a full example I use to test whether it works? I tried a simple example but that didn't cause the error

Reply:
Here is an example...

```rust
#[kani::ensures(result.is_some())]
fn or_default<T: Default>(opt: Option<T>) -> Option<T> {
    opt.or(Some(T::default()))
}

#[kani::proof_for_contract(or_default)]
fn harness() {
    let input: Option<i32> = kani::any();
    or_default(input);
}
```

Reply:
Thanks. Has been implemented. It does require a `kani::Arbitrary` bound on `T` though.